File: sample/dracula.kt
data class Vampire(
    val location: String,
    val birthDate: Int,
    val deathDate: Int,
    val weaknesses: List<String>,
) {
    val age: Int
        get() = deathDate - birthDate
}

fun main() {
    val dracula = Vampire("Transylvania", 1428, 1476, listOf("Sunlight", "Garlic"))
    val alucard = Vampire("Wallachia", 1458, 1510, listOf("Holy Water"))

    println("Dracula: ${dracula.age}, Alucard: ${alucard.age}")
}

File: sample/dracula.rb
Vampire = Data.define(:location, :birth_date, :death_date, :weaknesses) do
  def age
    death_date - birth_date
  end
end

dracula = Vampire.new("Transylvania", 1428, 1476, %w[Sunlight Garlic])
alucard = Vampire.new("Wallachia", 1458, 1510, ["Holy Water"])

puts({ dracula_age: dracula.age, alucard_age: alucard.age })
